### PR TITLE
Support mapped types and generics

### PR DESCRIPTION
**Pull request by wesleyyee**

### CODE REVIEW

**Comment by wesleyyee** [comment]
@googlebot I signed it!

**Comment by wesleyyee** [comment]
@googlebot I fixed it.

**Comment by wesleyyee** [comment]
@googlebot I fixed it.

File: src/lang/ts/intermock.ts [hunk 1]
```diff
@@ -189,6 +189,30 @@ function processFunctionPropertyType(
...
 }
 
+function processIndexedAccessPropertyType(
+    node: ts.IndexedAccessTypeNode, output: Output, property: string,
+    options: Options, types: Types) {
+  const objectType =
+      ((node.objectType as ts.TypeReferenceNode).typeName as ts.Identifier)
+          .escapedText;
+  const indexType =
+      ((node.indexType as ts.LiteralTypeNode).literal as any).text;
```
Comment (ryanmcdermott) [inline src/lang/ts/intermock.ts]: Can you remove the use of `any`? It loses type safety!

**Comment by ryanmcdermott** [comment]
This is amazing! Thank you so much for your work!

**Comment by wesleyyee** [comment]
@ryanmcdermott now that the PR is merged can we bump the version?

**Pull request merged**